File: src/Entity/Cours.php
<?php

namespace App\Entity;
use Symfony\Component\Validator\Constraints as Assert;
use App\Repository\CoursRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Annotation\Groups;

/**
 * @ORM\Entity(repositoryClass=CoursRepository::class)
 */
class Cours
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     * @Groups("post:read")

     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank
     * @Groups("post:read")
     */
    private $nom;

    /**
     * @Assert\NotBlank(message="le nom du coach  doit etre non vide")
     * @ORM\Column(type="string", length=255)
     * @Assert\Type(type="alpha", message="Le nom du coach ne doit pas contenir des chiffres .")
     * @Groups("post:read")
     */
    private $nomCoach;


    /**
     * @ORM\Column(type="integer")
     * @Assert\NotBlank
     * @Assert\Positive(message="le nombre doit etre positif")
     * @Groups("post:read")
     */
    private $nombre;

    /**
     * @ORM\Column(type="string", length=500)
     *  @Assert\NotBlank
     * @Groups("post:read")
     */
    private $description;

    /**
     * @ORM\Column(type="string", length=255)
     * @Groups("post:read")
     */
    private $image;



    /**
     * @ORM\Column(type="time", nullable=true)
     * @Groups("post:read")
     */
    private $heureD;

    /**
     * @ORM\Column(type="time", nullable=true)
     * @Groups("post:read")
     */
    private $heureF;

    /**
     * @ORM\ManyToOne(targetEntity=Salle::class, inversedBy="cours")
     * @ORM\JoinColumn(nullable=false)
     *  @Assert\NotBlank(message="il faut choisir une salle de sport")
     * @Groups("post:read")
     */
    private $salleassocie;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank
     * @Groups("post:read")
     */
    private $jour;




    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getNomCoach(): ?string
    {
        return $this->nomCoach;
    }

    public function setNomCoach(string $nomCoach): self
    {
        $this->nomCoach = $nomCoach;

        return $this;
    }



    public function getNombre(): ?int
    {
        return $this->nombre;
    }

    public function setNombre(int $nombre): self
    {
        $this->nombre = $nombre;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(string $image): self
    {
        $this->image = $image;

        return $this;
    }
    public function __toString()
    {
        return (string) $this->getSalleassocie();
    }




    public function getHeureD(): ?\DateTimeInterface
    {
        return $this->heureD;
    }

    public function setHeureD(\DateTimeInterface $heureD): self
    {
        $this->heureD = $heureD;

        return $this;
    }

    public function getHeureF(): ?\DateTimeInterface
    {
        return $this->heureF;
    }

    public function setHeureF(\DateTimeInterface $heureF): self
    {
        $this->heureF = $heureF;

        return $this;
    }

    public function getSalleassocie(): ?Salle
    {
        return $this->salleassocie;
    }

    public function setSalleassocie(?Salle $salleassocie): self
    {
        $this->salleassocie = $salleassocie;

        return $this;
    }

    public function getJour(): ?string
    {
        return $this->jour;
    }

    public function setJour(string $jour): self
    {
        $this->jour = $jour;

        return $this;
    }













}

File: Mobile/ENERGYM/src/com/mycompany/entities/Cours.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.entities;

/**
 *
 * @author nouri
 */
public class Cours {
     private float id;
     private String nom;
     private String nomCoach;
     private String description;
     private String nombre;
     private String image;
     private String heureD;
     private String heureF;
     private String salleassocie;
     private String jour;

    public Cours() {
    }

    public Cours(float id, String nom, String nomCoach, String description, String nombre, String jour) {
        this.id = id;
        this.nom = nom;
        this.nomCoach = nomCoach;
        this.description = description;
        this.nombre = nombre;
        this.jour = jour;
    }

    public Cours(String nom, String nomCoach, String description, String nombre, String salleassocie, String jour) {
        this.nom = nom;
        this.nomCoach = nomCoach;
        this.description = description;
        this.nombre = nombre;
        this.salleassocie = salleassocie;
        this.jour = jour;
    }
   

    public float getId() {
        return id;
    }

    public String getNom() {
        return nom;
    }

    public String getNomCoach() {
        return nomCoach;
    }

    public String getDescription() {
        return description;
    }

    public String getNombre() {
        return nombre;
    }

    public String getImage() {
        return image;
    }

    public String getHeureD() {
        return heureD;
    }

    public String getHeureF() {
        return heureF;
    }

    public String getSalleassocie() {
        return salleassocie;
    }

    public String getJour() {
        return jour;
    }

    public void setId(float id) {
        this.id = id;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public void setNomCoach(String nomCoach) {
        this.nomCoach = nomCoach;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public void setHeureD(String heureD) {
        this.heureD = heureD;
    }

    public void setHeureF(String heureF) {
        this.heureF = heureF;
    }

    public void setSalleassocie(String salleassocie) {
        this.salleassocie = salleassocie;
    }

    public void setJour(String jour) {
        this.jour = jour;
    }
   
     
}